### ISSUE TITLE
DynamicField Database: Filter for DF doesn't work in AgentTicketFreeText (but works in AgentTicketPhone)

### ISSUE THREAD

**Issue by StefanAbel-OTOBO** [RotherOSS/otobo#2165]
When having two DynamicField Database, you can use the first one to limit the results of the second one.

This works in AgentTicketPhone:
![image](https://user-images.githubusercontent.com/87359203/218430473-5b8ac445-e400-4401-9704-59abad8f05d0.png)

This doesn't work in AgentTicketFreeText, as you don't see any result (too much limitation):
![image](https://user-images.githubusercontent.com/87359203/218430660-8b9d0763-5cf0-4f27-b48b-7e9a8488c1fa.png)

If you remove the Filter for DynamicField Agent1 in the settings for Agent2, you see everything:
![image](https://user-images.githubusercontent.com/87359203/218430905-d4cda250-e272-49c3-aaae-7cc3d24d2a1d.png)

This is the configuration of the DynamicFields:
Agent1:
![image](https://user-images.githubusercontent.com/87359203/218430352-b28e9e58-dc3f-4912-891f-d043090ce023.png)

Agent2:
![image](https://user-images.githubusercontent.com/87359203/218430410-ba7daa7b-8cd4-4373-b7d9-e3946a992531.png)

I had this on a customer's system and could reproduce it in 10.1.6

**Comment by StefanAbel-OTOBO** [comment]
ov147

**Comment by stefanhaerter** [comment]
**TL;DR:** The bug-causing behavior looks like to be implemented on purpose. Further discussion is needed before changing this.

I did some additional debugging to verify that my in #2198 proposed solution is valid. I found that the behavior actually appears to be on purpose, at least a little bit: If a ticket id is given,  `Kernel::System::DynamicFieldDB::DatabaseSearchByConfig()` uses `Kernel::System::DynamicField::Backend::ValueGet()` <sup>1</sup> to retrieve the **stored** value of the given field for the ticket to perform its restrictions. In AgentTicketPhone, no ticket id exists yet, so the value given within the Params (and previously fetched by `EditFieldValueGet()`) is used. In AgentTicketFreeText, there exists a ticket id and even if the stored value is empty, it is used over the one given in the Params<sup>1</sup>.

This can easily be changed, but I would categorize this as a non-trivial design decision with at least a bit of impact, thus I would prefer discussing this with @svenoe and @bschmalhofer.

<sup>1</sup>

https://github.com/RotherOSS/otobo/blob/312e01a70186158f3c8dea3fe6df56ce11a7a017/Kernel/System/DynamicFieldDB.pm#L215-L219

**Comment by stefanhaerter** [comment]
Shifting this to 10.0 because a decision is to be made if the change in behavior is wanted or not.

**Comment by stefanhaerter** [comment]
Following a brief discussion, this issue is put into OTOBO Issues project for planning.
